### PR TITLE
Create Go SDK Documentation

### PR DESCRIPTION
**Pull request by NMCarv**
Created initial base for Go SDK documentation.

- created quickstart guide
- created page + snippets for data assets
- created page + snippets for data models

TODO: example repository (to check with @Siddharth9890)

### CODE REVIEW

File: developer-guide/go-sdk/data-models.mdx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+               "description": "Employee's department",
+           },
+           "profilePhoto": map[string]interface{}{
+               "type": "string",
+               "title": "Profile Photo",
+               "description": "Employee's profile picture",
+               "contentMediaType": "image/png",
+           },
+       },
+       "additionalProperties": false,
```
Comment (Siddharth9890) [inline developer-guide/go-sdk/data-models.mdx:62]: same as js sdk please remove this

File: developer-guide/go-sdk/data-models.mdx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+model, err := client.DataModel.GetById(modelID)
+if err != nil {
+    log.Fatalf("Failed to get data model: %v", err)
+}
+fmt.Printf("Retrieved model: %s (ID: %d)\n", model.Title, model.Id)
+
+// List Data Models created by the current user (paginated)
+page := 1
+pageSize := 10
+myModels, err := client.DataModel.GetCreatedByMe(page, pageSize)
```
Comment (Siddharth9890) [inline developer-guide/go-sdk/data-models.mdx:152]: same we dont have this in api

File: developer-guide/go-sdk/quickstart.mdx [hunk 3]
```diff
@@ -2,4 +2,110 @@
...
+```bash
+go get github.com/Gateway-DAO/gateway-go-sdk
+```
+
+## Authentication
+
+The SDK supports multiple authentication methods:
+
+- JWT-based authentication
+- Wallet-based authentication (Ethereum, Solana)
```
Comment (Siddharth9890) [inline developer-guide/go-sdk/quickstart.mdx:20]: add Sui

**Review by Siddharth9890 — CHANGES_REQUESTED** [review]

File: developer-guide/go-sdk/data-models.mdx (hunk 1 above)
Comment (NMCarv) [reply developer-guide/go-sdk/data-models.mdx:62]: We can technically pass this to the API (as part of the schema), and the `schema` on the SDK is `map[string]interface{}`. We'll be fine!

File: developer-guide/go-sdk/data-models.mdx (hunk 2 above)
Comment (NMCarv) [reply developer-guide/go-sdk/data-models.mdx:152]: We can replace with `GetMy`

**Review by Siddharth9890 — APPROVED** [review]

**Pull request merged**